### DISCUSSION TITLE
Jacobians

### DISCUSSION THREAD

**Discussion by Hanquist** [mutationpp/Mutationpp#274] [Ideas]
We are interested in using the Jacobians so we can leverage the implicit solver with the Mutation coupling to SU2. We could implement them on the SU2 side but I would prefer to do them in Mutation so others can leverage them as well. 

1. I don't think the Jacobians are in Mutation yet?
2. Where would be a good place to start?

**Comment by mgoodson-cvd** [comment]
Mutation++ currently provides an analytic Jacobian of chemical production rates with respect to the species densities: $`\frac{\partial \dot{\omega}}{\partial \rho_i}`$. The function is named `jacobianRho`; see [Kinetics.h](https://github.com/mutationpp/Mutationpp/blob/63ac65d9b3f95d9297b909165a53081505e5d1a7/src/kinetics/Kinetics.h#L195) for more details on ordering of entries (which is really important).

Note that Jacobians are not provided for chemical reaction rates with respect to any other quantity (e.g., temperature).  That is an open issue, #129.

Additionally, if running in thermal non-equilibrium (i.e., two temperatures), no Jacobians are provided for the energy exchange source term. If your solver needs this, a numerical Jacobian using a finite difference approach is the easiest solution for now.

**Reply by Hanquist** [reply]
Thanks for this insight @mgoodson-cvd! That is what I thought regarding there were some available but not all of them, at least for our needs. @aaronlarsen will be leading the effort here on it. 

We will eventually want to run in thermal nonequilibrium but we can start with 1T.

I will leave this discussion open as we work on it as I am sure we will have questions and also welcome any help.
